src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -144,6 +143,21 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   {
     commonAngles << QPair<double, QString>( *it, formatCommonAngleSnapping( *it ) );
   }
+
+  {
+    QAction *action = new QAction( tr( "Snapping to vertex/segment has priority over common angles" ), mCommonAngleActionsMenu );
```

Comment:
Again, this should not be limited to vertex/segment, rather to snapping in general. Not sure how exactly the wording could be simplified

Reply:
Maybe I could invert the logic (default "checked"): 
[ X ] Snapping to features has priority over common angle.

This way we don't change the current behavior by default.

Reply:
done, see updated snapshot